src/components/Layout/Header/ActionCenter/components/Notifications/SendNotification.tsx

```diff
@@ -0,0 +1,73 @@
+import { usePrevious } from '@chakra-ui/react'
+import type { RenderProps } from '@chakra-ui/react/dist/types/toast/toast.types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useEffect, useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { GenericTransactionNotification } from './GenericTransactionNotification'
+
+import { useTxStatus } from '@/hooks/useTxStatus/useTxStatus'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import { selectWalletActionsSorted } from '@/state/slices/actionSlice/selectors'
+import { ActionStatus, isGenericTransactionAction } from '@/state/slices/actionSlice/types'
+import { useAppSelector } from '@/state/store'
+
+export type GenericTransactionNotificationProps = {
+  handleClick: () => void
+  actionId: string
+} & RenderProps
+
+export const SendNotification = ({
+  handleClick,
+  actionId,
+  onClose,
+}: GenericTransactionNotificationProps) => {
+  const dispatch = useDispatch()
+  const actions = useAppSelector(selectWalletActionsSorted)
+  const action = useMemo(
+    () => actions.filter(isGenericTransactionAction).find(a => a.id === actionId),
+    [actions, actionId],
+  )
+
+  const transactionMetadata = useMemo(
+    () => action?.transactionMetadata,
+    [action?.transactionMetadata],
+  )
```

Comment:
Hey, why not but maybe a bit too much? why not consuming it directly? Is there too much rerenders for some reasons?

Reply:
dis gone